tests/EStimLibrary.UnitTests/Core/HardwareInterfaces/NeuralInterfaceManagerTests.cs

```diff
@@ -0,0 +1,207 @@
+    /// </summary>
+    /// <param name="interfaceType">The type of the hardware interface, for example: ContactGroup, GelPad</param>
+    /// <param name="interfaceSpecificParams">Parameters specific to that interface</param>
+    [Theory]
+    [InlineData(typeof(ContactGroup), new object[]{1})]
+    [InlineData(typeof(GelPad), new object[]{})]
+    public void CreateAndRegisterNeuralInterface_ShouldAssignUniqueGlobalContactIDs_WhenFirstInterface(Type interfaceType, object[] interfaceSpecificParams)
+    {
+        // Arrange
+        var NIManager = new NeuralInterfaceManager();
+
+        // Act
+        var contactIds = NIManager.CreateAndRegisterNeuralInterface(interfaceType, interfaceSpecificParams, out _);
+
+        // Assert
+        Assert.Equal(contactIds, contactIds.Distinct());
+    }
+
+    /// <summary>
+    /// The manager assigns global contact IDs for the new neural interface. 
+    /// Test that this contact assignment upholds global contact ID uniqueness when the interface is not the first one added.
+    /// </summary>
+    /// <param name="interfaceType">The type of the hardware interface, for example: ContactGroup, GelPad</param>
+    /// <param name="interfaceSpecificParams">Parameters specific to that interface</param>
+    [Theory]
+    [InlineData(typeof(ContactGroup), new object[]{1})]
+    [InlineData(typeof(GelPad), new object[]{})]
+    public void CreateAndRegisterNeuralInterface_ShouldAssignUniqueGlobalContactIDs_WhenNotFirstInterface(Type interfaceType, object[] interfaceSpecificParams)
+    {
+        // Arrange
+        var NIManager = new NeuralInterfaceManager();
+
+        // Act
+        var contactIds1 = NIManager.CreateAndRegisterNeuralInterface(interfaceType, interfaceSpecificParams, out _);
+        var contactIds2 = NIManager.CreateAndRegisterNeuralInterface(interfaceType, interfaceSpecificParams, out _);
+
+        // Assert
+        var allContactIds = contactIds1.Union(contactIds2).ToList();
+        Assert.Equal(allContactIds.Count, allContactIds.Distinct().Count());
+    }
+
+    /// <summary>
+    /// The manager assigns global contact IDs for the new neural interface. 
+    /// Test that the correct number of contact IDs was assigned.
+    /// </summary>
+    /// <param name="numContacts">Number of contacts for the Neural Interface</param>
+    [Theory]
+    [InlineData(1)]
+    [InlineData(5)]
+    [InlineData(10)]
+    public void CreateAndRegisterNeuralInterface_ShouldAssignCorrectNumberOfContactIDs(int numContacts)
+    {
+        // Arrange
+        var NIManager = new NeuralInterfaceManager();
+        Type interfaceType = typeof(ContactGroup);
+        object[] parameters = new object[] { numContacts };
+
+        // Act
+        var contactIds = NIManager.CreateAndRegisterNeuralInterface(interfaceType, parameters, out _);
+
+        // Assert
+        Assert.Equal(numContacts, contactIds.Count);
+    }
+
```

Comment:
Also may want to consider some test cases that add more than one interface to the manager at once! And test adding an interface with a single contact first followed by a multi-contact second interface, or vice versa